### PR TITLE
Add tests for parsing search results to reflectometry GUI

### PR DESCRIPTION
**Pull request by gemmaguest**
This PR adds unit tests for constructing and parsing the search results for the ISIS Reflectometry GUI. It moves the code to do the parsing into the `SearchResult` class from the `RunsPresenter`.

**To test:**

- Code review and ensure unit tests pass
- Open the ISIS Reflectometry interface
- Set the instrument to INTER and enter investigation ID `1120015` and click Search. Log in to ICat if prompted.
- Select some of the results e.g. the first 7 rows and click transfer. They should either be transferred to the table or highlighted in blue to indicate an error. Hover over the blue rows to see a tooltip of the error message.
- Delete all rows in the table and add the same (valid) ones again but in reverse order. They should be sorted so that you get the same result as before.
- Try transferring the same row twice. It should only be added once.

image

Refs #26512

*This does not require release notes* because **it concerns testing only**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by NickDraper** [comment]
This has moved around some GUI code so it needs some manual testing as well.  Please add some instructions into the to test section.

**Review by NickDraper — COMMENTED** [review]
Code review is good, just needs some manual testing as well

**Comment by gemmaguest** [comment]
> This has moved around some GUI code so it needs some manual testing as well. Please add some instructions into the to test section.

@NickDraper I've added the instructions

**Comment by NickDraper** [comment]
This all works fine

**Review by NickDraper — DISMISSED** [review]

File: qt/scientific_interfaces/ISISReflectometry/GUI/Runs/SearchResult.cpp [hunk 1]
```diff
@@ -5,14 +5,51 @@
...
+std::string SearchResult::description() const { return m_description; }
+
+std::string SearchResult::location() const { return m_location; }
+
+std::string SearchResult::error() const { return m_error; }
+
+std::string SearchResult::groupName() const { return m_groupName; }
+
+std::string SearchResult::theta() const { return m_theta; }
+
```
Comment (quantumsteve) [inline qt/scientific_interfaces/ISISReflectometry/GUI/Runs/SearchResult.cpp]: Should these 6 member function instead return `const std::string &`, which better matches the argument for ([QString::fromStdString]https://doc.qt.io/qt-5/qstring.html#fromStdString)?

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Can we avoid temporary strings or are there lifetime issues?

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**